contracts/limitOrder/LimitOrderBook.sol

```diff
@@ -0,0 +1,170 @@
+        );
+
+    //
+    // EXTERNAL NON-VIEW
+    //
+
+    function initialize(
+        string memory name,
+        string memory version,
+        address clearingHouseArg,
+        address limitOrderFeeVaultArg
+    ) external initializer {
+        __ReentrancyGuard_init();
+        __OwnerPausable_init();
+        __EIP712_init(name, version); // ex: "PerpCurieLimitOrder" and "1"
+
+        // LOB_CHINC : ClearingHouse Is Not Contract
+        require(clearingHouseArg.isContract(), "LOB_CHINC");
+        clearingHouse = clearingHouseArg;
+        accountBalance = IClearingHouse(clearingHouse).getAccountBalance();
+
+        // LOB_LOFVINC : LimitOrderFeeVault Is Not Contract
+        require(limitOrderFeeVaultArg.isContract(), "LOB_LOFVINC");
+        limitOrderFeeVault = limitOrderFeeVaultArg;
+    }
+
+    function setClearingHouse(address clearingHouseArg) external onlyOwner {
+        // LOB_CHINC: ClearingHouse Is Not a Contract
+        require(clearingHouseArg.isContract(), "LOB_CHINC");
+        clearingHouse = clearingHouseArg;
+        accountBalance = IClearingHouse(clearingHouse).getAccountBalance();
+
+        emit ClearingHouseChanged(clearingHouseArg);
+    }
+
+    function setLimitOrderFeeVault(address limitOrderFeeVaultArg) external onlyOwner {
+        // LOB_LOFVINC: LimitOrderFeeVault Is Not a Contract
+        require(limitOrderFeeVaultArg.isContract(), "LOB_LOFVINC");
+        limitOrderFeeVault = limitOrderFeeVaultArg;
+
+        emit LimitOrderFeeVaultChanged(limitOrderFeeVaultArg);
+    }
+
+    /// @inheritdoc ILimitOrderBook
+    function fillLimitOrder(
+        LimitOrder memory order,
+        bytes memory signature,
+        uint80 roundIdWhenTriggered
+    ) external override nonReentrant {
+        bytes32 orderHash = getOrderHash(order);
+        verifySigner(order, signature);
+
+        // TODO: support StopLimitOrder in the future
+        // LOB_OSLO: Only Support LimitOrder
+        require(order.orderType == ILimitOrderBook.OrderType.LimitOrder, "LOB_OSLO");
+
+        // LOB_OIFA: Order is filled already
+        require(_ordersStatus[orderHash] != ILimitOrderBook.OrderStatus.Filled, "LOB_OIFA");
+        // LOB_OIC: Order is cancelled
+        require(_ordersStatus[orderHash] != ILimitOrderBook.OrderStatus.Cancelled, "LOB_OIC");
+
+        int256 oldTakerPositionSize = IAccountBalance(accountBalance).getTakerPositionSize(
+            order.trader,
+            order.baseToken
+        );
+
+        (uint256 base, uint256 quote) = IClearingHouse(clearingHouse).openPositionFor(
+            order.trader,
+            IClearingHouse.OpenPositionParams({
+                baseToken: order.baseToken,
+                isBaseToQuote: order.isBaseToQuote,
+                isExactInput: order.isExactInput,
+                amount: order.amount,
+                oppositeAmountBound: order.oppositeAmountBound,
+                deadline: order.deadline,
+                sqrtPriceLimitX96: 0,
```

Comment:
have we considered to leverage this parameter, `sqrtPriceLimitX96` in limit order? If traders  can set this value properly, their order could be partial filled instead of reverted when they have big orders.

Reply:
not sure how to use `sqrtPriceLimitX96` for partial filled but it might increase more complexity based on the current spec. We could discuss this more on gather or meetings and prefer not add this feature in the current release

Reply:
actually don't need to do anything from contract side, what we need to do is have this parameter that can be passed from outside. Just for flexibility.

Reply:
Will we need to record how much is partial filled? I suppose there could be multiple filled events for an order if it's partial filled multiple time?

What will happen if keeper call `fillLimitOrder()` on a previously-partial-fiilled order?

Reply:
My idea is simpler, `sqrtPriceLimitX96` will be set by traders. So if they set this value, their order will be partial filled if it fits the criteria. The state of this order would become `Filled` which means no more actions for this order. 

It's all offchain behaviors, we don't need to decide how it will work now. We can support this parameter first and still set `0` in our service. Everything will work just like what we have now. And we could activate this feature in the future after we discussed the details.